uni-stark/src/prover.rs

```diff
@@ -90,11 +121,22 @@ where
     // a soundness issue.
     let alpha: SC::Challenge = challenger.sample_algebra_element();
 
+    // A domain large enough to uniquely identify the quotient polynomial.
+    // This domain must be contained in the domain over which `trace_data` is defined.
+    // Explicitly it should be equal to `gK` for some subgroup `K` contained in `H'`.
     let quotient_domain =
         ext_trace_domain.create_disjoint_domain(1 << (log_ext_degree + log_quotient_degree));
 
+    // Return a the subset of the extended trace `ET` corresponding to the rows giving evaluations
+    // over the quotient domain.
+    //
+    // This only works if the trace domain is `gH'` and the quotient domain is `gK` for some subgroup `K` contained in `H'`.
+    // TODO: Make this explicit in `get_evaluations_on_domain` or otherwise fix this.
     let trace_on_quotient_domain = pcs.get_evaluations_on_domain(&trace_data, 0, quotient_domain);
 
+    // Compute the quotient polynomial `Q(x)` by evaluating `C(T_1(x), ..., T_w(x), T_1(hx), ..., T_w(hx), selectors(x)) / Z_H(x)`
+    // at every point in the quotient domain. The degree of `Q(x)` is `<= deg(C) * (N - 1) - N + 1 = 2(N - 1)`.
+    // The `-N` comes from dividing by `Z_H(x)` and the `+1` is due to the `is_transition` selector.
```

Comment:
Maybe it's worth emphasizing that `is_first_row` and `is_last_row` are already captured by `deg(C)`, while `is_transition` is not (hence why we only correct for that selector here)?